x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/ActionTypes.java

```diff
@@ -23,4 +24,6 @@ public final class ActionTypes {
     );
 
     public static final ActionType<QueryUserResponse> QUERY_USER_ACTION = new ActionType<>("cluster:admin/xpack/security/user/query");
+
+    public static final ActionType<BulkPutRolesResponse> BULK_PUT_ROLES = new ActionType<>("cluster:admin/xpack/security/roles/put");
```

Comment:
Why do we need this here, and not be next to the `PutRoleAction`?
Separately, I would suggest to name this `cluster:admin/xpack/security/role/bulk_put`. Having actions that are named `security/role` and `security/roles` is a source of confusion, I think.

Reply:
I think you need this because the transport request builder must be in the core plugin in order to be able to inject a different one for serverless for role descriptor validation. Again, my suggestion, would be to not think about that in this PR.

Reply:
>Why do we need this here, and not be next to the PutRoleAction?

My understanding is that all new actions should be added to `ActionTypes` instead of creating separate subclasses of `ActionType`. `PutRoleAction ` is also defined in core, but not in the `ActionTypes` class. The purposes is to map the action in core to the transport action defined in the security plugin.

>Separately, I would suggest to name this cluster:admin/xpack/security/role/bulk_put. Having actions that are named security/role and security/roles is a source of confusion, I think.

Yes, I like that, I've updated it. 

>Again, my suggestion, would be to not think about that in this PR.

I think thinking about this is very minimal (only the request builder), so I rather keep it.